app/services/file_locator.rb

```diff
@@ -30,6 +30,14 @@ def initialize(uri)
     def object
       @object ||= Aws::S3::Object.new(bucket_name: bucket, key: key)
     end
+
+    def localize(dir = "/tmp")
+      new_dir = File.join dir, SecureRandom.uuid
+      new_path = File.join new_dir, File.basename(key)
+      FileUtils.mkdir_p new_dir
```

Comment:
What about using Tempfile instead of manually handling it in `/tmp`?  Maybe also add a method like `destroy_local_copy`?
https://ruby-doc.org/stdlib-2.4.0/libdoc/tempfile/rdoc/Tempfile.html#class-Tempfile-label-Explicit+close

Reply:
Auto delete would be nice but we don't know when ffmpeg would be done with the file. How would you implement that?